stan/math/version.hpp

```diff
@@ -1,8 +1,34 @@
 #ifndef STAN_MATH_VERSION_HPP
 #define STAN_MATH_VERSION_HPP
 
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <boost/version.hpp>
+#include <sundials/sundials_config.h>
 #include <string>
 
+#if __has_include(<tbb/tbb_stddef.h>)
+#include <tbb/tbb_stddef.h>
+#else
+#include <tbb/version.h>
+#endif
+
+#if EIGEN_VERSION_AT_LEAST(3, 4, 0)
```

Comment:
This looks like it'll be useful, but maybe not done this way? I'm not sure if this is where we should check. Maybe it is. Can you walk through the logic?